terraform/modules/cf_hosting/cloudfront.tf

```diff
@@ -18,7 +23,94 @@ resource "aws_cloudfront_distribution" "s3_distribution" {
 #   aliases = ["mysite.example.com", "yoursite.example.com"]
 
 	default_cache_behavior {
-		allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+		allowed_methods  = ["GET", "HEAD", "OPTIONS"]
+		cached_methods   = ["GET", "HEAD"]
+		target_origin_id = "${var.tags.org_name}-${var.tags.application_name}-${var.tags.environment_name}-orgin"
+
+		forwarded_values {
+			query_string = false
+			headers      = ["CloudFront-Viewer-Country", "x-nice-allowed-ip"]
+
+			cookies {
+				forward = "none"
+			}
+		}
+		lambda_function_association {
+			event_type = "origin-request"
+			lambda_arn = var.origin_request_edge_lambda_qualified_arn
+		}
+		lambda_function_association {
+			event_type = "origin-response"
+			lambda_arn = var.origin_repsonse_edge_lambda_qualified_arn
+		}
+		lambda_function_association {
+			event_type = "viewer-request"
+			lambda_arn = var.viewer_request_edge_lambda_qualified_arn
+		}
+
+		compress               = true
+		viewer_protocol_policy = "redirect-to-https"
+	}
+
+	ordered_cache_behavior {
```

Comment:
I think these would be useful to have a comment explaining why we have a cache behaviour for each static file type - to basically explain that we only need to IP restrict HTML pages, which mean assets are faster to serve and cheaper

Reply:
explain why....